matchzoo/preprocessors.py

```diff
@@ -195,6 +195,8 @@ def transform(self, tokens: list, ngram: int=3) -> list:
 
         :param tokens: list of tokens to be transformed.
         :param ngram: By default use 3-gram (tri-letter).
+
+        :return: set of tri-letters, dependent on `ngram`.
```

Comment:
In the current preprocessors, all operations are conducted on raw texts. However, we have to map each word/n-letters to id, which is  the function of `Vocabulary`.  Thus, we need to design a `Vocabulary` class, whose `fit` construct a _word_dict and `transform` map list of string to list of ids.

Reply:
@faneshion fire up an issue for that?

Reply:
OK